File: Client/src/app/admin/sages/sage.ts
import {Book} from '../books/book';

export interface Sage {
  id?: string;
  name?: string;
  age?: number;
  city?: string;
  books?: Book[];
}

export interface SageUpdate {
  id?: string;
  name?: string;
  age?: number;
  city?: string;
  books?: number[];
}

File: web_lab3.Abstractions/Entities/Sage.cs
using System.Collections.Generic;

namespace Abstractions.Entities
{
    public class Sage : IEntity<int>
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public int Age { get; set; }

        public string City { get; set; }

        public List<Book> Books { get; set; } = new();
    }
}